server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v70/PopulateDefaultQualityGateTest.java

```diff
@@ -0,0 +1,143 @@
+import org.sonar.db.CoreDbTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PopulateDefaultQualityGateTest {
+  private final static long PAST = 10_000_000_000L;
+  private static final long NOW = 50_000_000_000L;
+  private static final String TABLE_ORGANIZATIONS = "organizations";
+  private static final String TABLE_QUALITY_GATES = "quality_gates";
+
+  private System2 system2 = new TestSystem2().setNow(NOW);
+
+  @Rule
+  public CoreDbTester db = CoreDbTester.createForSchema(PopulateDefaultQualityGateTest.class, "organizations.sql");
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  private PopulateDefaultQualityGate underTest = new PopulateDefaultQualityGate(db.database(), system2);
+
+
+  @Test
+  public void has_no_effect_if_table_is_empty() throws SQLException {
+    insertQualityGate(Uuids.createFast(), "Sonar way", true);
+
+    underTest.execute();
+
+    assertThat(db.countRowsOfTable("organizations")).isEqualTo(0);
+  }
+
+  @Test
+  public void should_populate_defaultQualityGate_column() throws SQLException {
+    String builtInQGUuid = Uuids.createFast();
+    insertQualityGate(builtInQGUuid, "Sonar way", true);
+    String orgUuid1 = Uuids.createFast();
+    String orgUuid2 = Uuids.createFast();
+    insertOrganization(orgUuid1);
+    insertOrganization(orgUuid2);
+
+    underTest.execute();
+
+    // all organizations have the builtIn quality gate
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid != '" + builtInQGUuid + "'")
+    ).isEqualTo(0);
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid = '" + builtInQGUuid + "'")
+    ).isEqualTo(2);
+
+    // updated_at must have been updated
+    assertThat(
+      db.countSql("select count(uuid) from organizations where updated_at = " + NOW )
+    ).isEqualTo(2);
+  }
+
+  @Test
+  public void is_reentrant() throws SQLException {
+    String builtInQGUuid = Uuids.createFast();
+    insertQualityGate(builtInQGUuid, "Sonar way", true);
+    insertOrganization(Uuids.createFast());
+    insertOrganization(Uuids.createFast());
+
+    underTest.execute();
+    underTest.execute();
+
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid != '" + builtInQGUuid + "'")
+    ).isEqualTo(0);
+
+    assertThat(
+      db.countSql("select count(uuid) from organizations where default_quality_gate_uuid = '" + builtInQGUuid + "'")
+    ).isEqualTo(2);
+  }
+
+  @Test
+  public void should_fail_with_ISE_when_no_builtIn() throws SQLException {
+    insertOrganization(Uuids.createFast());
+
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("Unable to find the builtin quality gate");
+
+    underTest.execute();
+  }
+
+  @Test
+  public void should_not_fail_if_there_is_multiple_builtin_qualitygates() throws SQLException {
```

Comment:
If it would have been me, I would fail fast in this case => This is a strange situation where it's hard to predict what will happen